forge-game/src/main/java/forge/game/card/Card.java

```diff
@@ -622,6 +622,9 @@ public boolean changeCardState(final String mode, final String customState, fina
 
                 boolean result = c.changeToState(c.backside ? CardStateName.Transformed : CardStateName.Original);
                 retResult = retResult || result;
+                if (cause != null && cause.isCraft()) { // retain cards crafted while transforming in exile
+                    c.retainPaidList(cause, "ExiledCards");
```

Comment:
is this always needed? in that case it should just be part of the GameAction call instead

Reply:
Good catch